### PR TITLE
Increase submodule usability

### PR DESCRIPTION
**Pull request by gipohl**
Minor changes in reference file calls and automated file name creation to ensure usability of processes in subworkflows.

### CODE REVIEW

File: modules/02_star.nf [hunk 1]
```diff
@@ -13,12 +13,12 @@ process STAR {
...
     output:
       tuple val("${name}"), file("${name}.bam"), emit: bams
-      file("software_versions.${task.process}.txt")
+      file("software_versions.STAR.txt")
 
     script:
     two_pass_mode_param = params.star_two_pass_mode ? "--twopassMode Basic" : ""
     """
-    STAR --genomeDir ${params.reference} ${two_pass_mode_param} ${params.additional_args} \
+    STAR --genomeDir ${reference} ${two_pass_mode_param} ${params.additional_args} \
```
Comment (priesgo) [inline modules/02_star.nf:22]: you are missing the definition of input parameter `val(reference)`

**Review by priesgo — CHANGES_REQUESTED** [review]
Minor issue

File: modules/02_star.nf (hunk 1 above)
Comment (gipohl) [reply modules/02_star.nf:22]: Added missing input parameter in latest commit

**Review by priesgo — APPROVED** [review]
LGTM

**Pull request merged**